src/modules/Shadow/Module.js

```diff
@@ -0,0 +1,75 @@
+module.exports = function Shadow(options, UI) {
+
+    var output;
+
+    // The function which is called on every draw.
+    function draw(input, callback, progressObj) {
+        progressObj.stop(true);
+        progressObj.overrideFlag = true;
+
+        var step = this;
+
+        let defaults = require('./../../util/getDefaults.js')(require('./info.json'));
+        let shadowDimension = options.shadowDimension || defaults.shadowDimension;
+        shadowDimension = shadowDimension.split(" ");
+        var firstValue = shadowDimension[0];
+        var secondValue = shadowDimension[1];
+        console.log(firstValue, secondValue);
+
+        const getPixels = require("get-pixels");
+
+        getPixels(input.src, function(err, pixels) {
+            console.log("nirav");
+            if (options.preProcess) pixels = options.preProcess(pixels); // Allow for preprocessing
+            var pixelsArray = [];
+            var i = 0;
+            var pixel = [];
+            for (var x = 0; x < pixels.shape[0]; x++) {
+                for (var y = 0; y < pixels.shape[1]; y++) {
```

Comment:
You don't have to do all of this here. You can do it in extraManipulation without the use of get-pixels. Just put extraManipulation inside draw.